src/lib/strapi.ts

```diff
@@ -279,52 +281,45 @@ export class Strapi {
    *
    * @param  {string} contentType - Content type's name pluralized
    * @param  {AxiosRequestConfig["params"]} params? - Filter and order queries
-   * @returns Promise<T>
+   * @returns Promise<StrapiResponse<T>>
    */
   public find<T>(
     contentType: string,
     params?: AxiosRequestConfig["params"]
-  ): Promise<T> {
-    return this.request<T>("get", `/${contentType}`, { params });
+  ): Promise<StrapiResponse<T>> {
+    return this.request<StrapiResponse<T>>("get", `/${contentType}`, {
+      params,
+    });
   }
 
   /**
    * Get a specific {content-type} entry
    *
    * @param  {string} contentType - Content type's name pluralized
    * @param  {string|number} id - ID of entry
-   * @returns Promise<T>
+   * @returns Promise<StrapiResponse<T>>
    */
-  public findOne<T>(contentType: string, id: string | number): Promise<T> {
-    return this.request<T>("get", `/${contentType}/${id}`);
-  }
-
-  /**
-   * Count {content-type} entries
-   *
-   * @param  {string} contentType - Content type's name pluralized
-   * @param  {AxiosRequestConfig["params"]} params? - Filter and order queries
-   * @returns Promise<T>
-   */
-  public count<T>(
+  public findOne<T>(
     contentType: string,
-    params?: AxiosRequestConfig["params"]
-  ): Promise<T> {
-    return this.request<T>("get", `/${contentType}/count`, { params });
+    id: string | number
+  ): Promise<StrapiResponse<T>> {
+    return this.request<StrapiResponse<T>>("get", `/${contentType}/${id}`);
   }
 
   /**
    * Create a {content-type} entry
    *
    * @param  {string} contentType - Content type's name pluralized
    * @param  {AxiosRequestConfig["data"]} data - New entry
-   * @returns Promise<T>
+   * @returns Promise<StrapiResponse<T>>
    */
   public create<T>(
     contentType: string,
     data: AxiosRequestConfig["data"]
-  ): Promise<T> {
-    return this.request<T>("post", `/${contentType}`, { data });
+  ): Promise<StrapiResponse<T>> {
```

Comment:
When creating you can actually ask the api to only return certain fields or relations might be interesting toallow it here (same on update/delete)

Reply:
Right! I'll check the RFC ;)

Reply:
@alexandrebodin Should be done now 👍🏼

Reply:
you can add the populate syntax too

Reply:
> you can add the populate syntax too

Done with latest commit 😄